tests/glsl-intrinsic/shader-subgroup/shader-subgroup-clustered.slang

```diff
@@ -0,0 +1,47 @@
+//TEST:SIMPLE(filecheck=CHECK_GLSL):  -allow-glsl -stage compute -entry computeMain -target glsl
+//TEST:SIMPLE(filecheck=CHECK_SPV):  -allow-glsl -stage compute -entry computeMain -target spirv
+
+// not testing hlsl due to missing impl
+// TEST:SIMPLE(filecheck=CHECK_HLSL): -allow-glsl -stage compute -entry computeMain -target hlsl
+// not testing cuda due to missing impl
+// TEST:SIMPLE(filecheck=CHECK_CUDA): -allow-glsl -stage compute -entry computeMain -target cuda 
+// not testing cpp due to missing impl
+// TEST:SIMPLE(filecheck=CHECK_CPP):  -allow-glsl -stage compute -entry computeMain -target cpp
+
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl
+//TEST(compute, vulkan):COMPARE_COMPUTE(filecheck-buffer=BUF):-vk -compute -entry computeMain -allow-glsl -emit-spirv-directly
+#version 430
+
+precision highp float;
+precision highp int;
+
+//TEST_INPUT:ubuffer(data=[0], stride=4):out,name=outputBuffer
+buffer MyBlockName2 
+{
+    uint data[];
+} outputBuffer;
+
+layout(local_size_x = 32) in;
+
+#include "./glsl-test-utils.slang"
+
+void computeMain()
+{
+    float v = 1.0f;
+    outputBuffer.data[0] = testFunctionIn2ParamPlus(subgroupClusteredAdd, 1, v, 1) &&
+                           testFunctionIn2ParamPlus(subgroupClusteredMul, 1, v, 1) &&
+                           testFunctionIn2ParamPlus(subgroupClusteredMin, 1, v, 1) &&
+                           testFunctionIn2ParamPlus(subgroupClusteredMax, 1, v, 1) &&
+                           testFunctionInInt2ParamPlus(subgroupClusteredAnd, 1, v, 1) &&
+                           testFunctionInInt2ParamPlus(subgroupClusteredOr, 1, v, 1) &&
+                           testFunctionInInt2ParamPlus(subgroupClusteredXor, 1, v, 1);
```

Comment:
In a unit test code, we should be able to individually enable and disable each and every test as we progressively implement functions. Your way to wrapping a group of tests in to a macro is not preferable.
The use of "if" and "for" are also not preferable in unit test. If possible avoid using any complex logics and unroll them as much as possible so that things are easy to read, and harder to make a mistake. If unit tests start having complexity, we will have to have another unit tests that test the unit tests, which becomes unmaintainable.